src/node.cc

```diff
@@ -1069,11 +1069,14 @@ InitializationResult InitializeOncePerProcess(int argc, char** argv) {
   V8::SetEntropySource(crypto::EntropySource);
 #endif  // HAVE_OPENSSL
 
+#if NODE_USE_V8_PLATFORM
   per_process::v8_platform.Initialize(
       static_cast<int>(per_process::cli_options->v8_thread_pool_size));
   V8::Initialize();
   performance::performance_v8_start = PERFORMANCE_NOW();
   per_process::v8_initialized = true;
```

Comment:
Of these four lines:

- `per_process::v8_platform.Initialize()` is a no-op anyway if `NODE_USE_V8_PLATFORM` is not set
- `V8::Initialize()` is not really logically tied to `NODE_USE_V8_PLATFORM`
- `performance::performance_v8_start` is something you’d probably still want to set in any case, or at least it won’t hurt
- and ditto for `per_process::v8_initialized`.

So I don’t think `NODE_USE_V8_PLATFORM` is the right guard here – it just happens to work for Electron, because it doesn’t want `V8::Initialize()` to be run here, and it doesn’t set `NODE_USE_V8_PLATFORM`, but that’s really only a coincidence.

I think in an earlier conversation I mentioned that it would be nice if `InitializeOncePerProcess()` took a flags argument that specified what exactly it should initialize and what not, and I still think that that’s the right choice here overall.

Reply:
that works! ill rework this to add some of those option :)

Reply:
@addaleax updated - is this about what you were thinking?